testplan/exporters/testing/pdf/renderers/entries/baseUtils.py

```diff
@@ -0,0 +1,161 @@
+import os
+import uuid
+import tempfile
+import numpy as np
+from reportlab.platypus import Image
+from reportlab.lib.units import inch
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plot
+
+
+"""Convert a MatPlot plot into an image readable in the pdf"""
+def export_plot_to_image(graph_plot):
+    filename = '{0}.png'.format(uuid.uuid4())
+    temp_path = tempfile.gettempdir()
+    image_pathname = os.path.join(temp_path, filename)
+    graph_plot.savefig(image_pathname)
+    image = Image(image_pathname)
+    return image
+
+
+"""Resize the image"""
+def format_image(image):
+    image.drawWidth = 4 * inch
+    image.drawHeight = 3 * inch
+    return image
+
+
+"""returns two lists of x and y coordinates from dictionary key by strings"""
+def get_xy_coords(data):
+    x_values = []
+    y_values = []
+
+    for coord in data:
+        x_values.append(coord['x'])
+        y_values.append(coord['y'])
+
+    return x_values, y_values
+
+
+"""returns the graph colour for the series specified"""
+def get_colour(series_options, series_name):
+    if series_options is None:
+        return None
+    if series_name in series_options:
+        if 'colour' in series_options[series_name]:
+            return series_options[series_name]['colour']
+    return None
+
+
+"""returns the axis labels from graph options"""
+def get_axis_labels(graph_options):
+    if graph_options is None:
+        return None, None
+    x_axis_label = None
+    y_axis_label = None
+    if 'xAxisTitle' in graph_options:
+        x_axis_label = graph_options['xAxisTitle']
+    if 'yAxisTitle' in graph_options:
+        y_axis_label = graph_options['yAxisTitle']
+
+    return x_axis_label, y_axis_label
+
+
+"""returns whether to display the legend on the graph"""
+def show_legend(graph_options):
+    if graph_options is None:
+        return False
+    if 'legend' in graph_options:
+        return graph_options['legend']
+
+    return False
+
+
+"""
+calls appropriate plotting function based on 
+whether a graph or chart is being plotted
+"""
+def get_matlib_plot(source):
+    graph_type = source['graph_type']
+
+    valid_graph_types = ['Line', 'Scatter', 'Bar',
+                         'Whisker', 'Contour', 'Hexbin']
+    valid_chart_types = ['Pie']
+
+    if graph_type in valid_graph_types:
+        return plot_graph(source, graph_type)
+    elif graph_type in valid_chart_types:
+        return plot_chart(source, graph_type)
+    else:
+        return None
+
+
+"""
+Creates MatPlot plot for any graph requiring 
+axis that can use get_xy_coords function
+"""
+def plot_graph(source, graph_type):
+    data = source['graph_data']
+    graph_options = source['graph_options']
+    series_options = source['series_options']
+
+    # Special logic for multi-bar graph
+    if graph_type is 'Bar':
```

Comment:
Use `==` rather than `is` for string comparison. `is` checks that the objects themselves are the same which can sometimes produce surprising results: https://stackoverflow.com/questions/1504717/why-does-comparing-strings-using-either-or-is-sometimes-produce-a-differe